src/aws-healthomics-mcp-server/awslabs/aws_healthomics_mcp_server/search/genomics_search_orchestrator.py

```diff
@@ -0,0 +1,1145 @@
+        if request.file_type:
+            from awslabs.aws_healthomics_mcp_server.models import GenomicsFileType
+
+            try:
+                GenomicsFileType(request.file_type.lower())
+            except ValueError:
+                valid_types = [ft.value for ft in GenomicsFileType]
+                raise ValueError(
+                    f"Invalid file_type '{request.file_type}'. Valid types: {valid_types}"
+                )
+
+        logger.debug(f'Search request validation passed: {request}')
+
+    async def _execute_parallel_searches(
+        self, request: GenomicsFileSearchRequest
+    ) -> List[GenomicsFile]:
+        """Execute searches across all configured storage systems in parallel.
+
+        Args:
+            request: Search request containing search parameters
+
+        Returns:
+            Combined list of GenomicsFile objects from all storage systems
+        """
+        search_tasks = []
+
+        # Add S3 search task if bucket paths are configured and S3 engine is available
+        if self.config.s3_bucket_paths and self.s3_engine is not None:
+            logger.info(f'Adding S3 search task for {len(self.config.s3_bucket_paths)} buckets')
+            s3_task = self._search_s3_with_timeout(request)
+            search_tasks.append(('s3', s3_task))
+
+        # Add HealthOmics search tasks if enabled
+        if self.config.enable_healthomics_search:
+            logger.info('Adding HealthOmics search tasks')
+            sequence_task = self._search_healthomics_sequences_with_timeout(request)
+            reference_task = self._search_healthomics_references_with_timeout(request)
+            search_tasks.append(('healthomics_sequences', sequence_task))
+            search_tasks.append(('healthomics_references', reference_task))
+
+        if not search_tasks:
+            logger.warning('No storage systems configured for search')
+            return []
+
+        # Execute all search tasks concurrently
+        logger.info(f'Executing {len(search_tasks)} parallel search tasks')
+        results = await asyncio.gather(*[task for _, task in search_tasks], return_exceptions=True)
+
+        # Collect results and handle exceptions
+        all_files = []
+        for i, result in enumerate(results):
+            storage_system, _ = search_tasks[i]
+            if isinstance(result, Exception):
+                logger.error(f'Error in {storage_system} search: {result}')
+                # Continue with other results rather than failing completely
+            elif isinstance(result, list):
+                logger.info(f'{storage_system} search returned {len(result)} files')
+                all_files.extend(result)
+            else:
+                logger.warning(f'Unexpected result type from {storage_system}: {type(result)}')
+
+        # Periodically clean up expired cache entries (approximately every 10th search)
+        if (
+            secrets.randbelow(10) == 0 and self.s3_engine is not None
```

Comment:
The same question about timing the cache cleanup: also how the 5% and 10% are determined